lib/chef/server_api_class_eval.rb

```diff
@@ -0,0 +1,64 @@
+# Author:: Jeremy Miller (<jmiller@chef.io>)
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Chef::ServerAPI.class_eval do
```

Comment:
Would it make more sense to do this metaprogramming inside the Error Handling file? At the end of the day we are adding an exception and also adding a few rescues to handle the errors from any request made with `ServerAPI`. Thoughts?